frontend/catalyst/pennylane_extensions.py

```diff
@@ -1201,6 +1209,9 @@ def __call__(self):
             assert mode == EvaluationMode.INTERPRETATION, f"Unsupported evaluation mode {mode}"
             return self._call_during_interpretation()
 
+    def _set_id(self, x):
```

Comment:
It is stange that it is not covered. Don't you need to add the same logic to `if_stmt`?

Reply:
Yes I was initially adding all the same logic to `if_stmt` but it turned up a bunch of difficult questions again when to raise, warn, or do something else. So I reverted those changes to not delay the release.
Note that currently `if_stmt` doesn't actually have a fallback to Python, if something goes wrong it always raises an error.